libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/oauth/_handlers/agentic_user_authorization.py

```diff
@@ -0,0 +1,148 @@
+import logging
+
+from typing import Optional
+
+from microsoft_agents.activity import TokenResponse
+
+from ....turn_context import TurnContext
+from ...._oauth import _FlowStateTag
+from .._sign_in_response import _SignInResponse
+from ._authorization_handler import _AuthorizationHandler
+from ....storage import Storage
+from ....authorization import Connections
+from ..auth_handler import AuthHandler
+
+logger = logging.getLogger(__name__)
+
+
+class AgenticUserAuthorization(_AuthorizationHandler):
+    """Class responsible for managing agentic authorization"""
+
+    def __init__(
+        self,
+        storage: Storage,
+        connection_manager: Connections,
+        auth_handler: Optional[AuthHandler] = None,
+        *,
+        auth_handler_id: Optional[str] = None,
+        auth_handler_settings: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        """
+        Creates a new instance of Authorization.
+
+        :param storage: The storage system to use for state management.
+        :type storage: Storage
+        :param connection_manager: The connection manager for OAuth providers.
+        :type connection_manager: Connections
+        :param auth_handlers: Configuration for OAuth providers.
+        :type auth_handlers: dict[str, AuthHandler], optional
+        :raises ValueError: When storage is None or no auth handlers provided.
+        """
+        super().__init__(
+            storage,
+            connection_manager,
+            auth_handler,
+            auth_handler_id=auth_handler_id,
+            auth_handler_settings=auth_handler_settings,
+            **kwargs,
+        )
+        self._alt_blueprint_name = auth_handler._alt_blueprint_name if auth_handler else None
+        
+
+    async def get_agentic_instance_token(self, context: TurnContext) -> TokenResponse:
+        """Gets the agentic instance token for the current agent instance.
+
+        :param context: The context object for the current turn.
+        :type context: TurnContext
+        :return: The agentic instance token, or None if not an agentic request.
+        :rtype: Optional[str]
+        """
+
+        if not context.activity.is_agentic_request():
+            return TokenResponse()
```

Comment:
SHouldn't this be an error? IF this ic called and it's not an agentic request, something is broken higher up in the stack.